### PR TITLE
Splink3 default schema

### PR DESCRIPTION
**Pull request by Th368MoJ**
Adjust our schema logic to simply use the default schema in our given backend.

For `duckdb`, the main database and any objects in our default schema can be accessed without the schema suffix.
e.g. these statements are both equivalent when a default schema of `splink` is set.
`linker.con.execute("SELECT * FROM __splink__cfac95d").fetch_df()`
`linker.con.execute("SELECT * FROM splink.__splink__cfac95d").fetch_df()`

Other options we have available to us in the future, if there's a specific SQL backend we can't use this for:
* A simple find and replace for items contained in our 
* Use sqlglot's parsing feature inside of our `_generate_pipeline` method, to apply the schema suffix to our required datasets.

### CODE REVIEW

File: splink/duckdb/duckdb_linker.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def __init__(
         input_tables={},
         connection=":memory:",
         set_up_basic_logging=True,
+        schema="",
```
Comment (RobinL) [inline splink/duckdb/duckdb_linker.py:46]: Could we call this `output_schema` just for clarity

File: splink/duckdb/duckdb_linker.py [hunk 2]
```diff
@@ -72,6 +73,13 @@ def __init__(
         input_tables = input_tables_new
 
         super().__init__(settings_dict, input_tables, set_up_basic_logging)
+        if schema:
+            self.con.execute(f"""
+            CREATE SCHEMA IF NOT EXISTS {schema};
+            SET schema '{schema}';
+            """)
+
+            print("Schema set to {schema}.\n")
```
Comment (RobinL) [inline splink/duckdb/duckdb_linker.py:82]: Delete this print statement?

File: splink/linker.py [hunk 3]
```diff
@@ -173,6 +175,11 @@ def two_dataset_link_only(self):
         else:
             return False
 
+    def _create_schema(self, table_name):
```
Comment (RobinL) [inline splink/linker.py:178]: Could we call this `_prepend_schema_to_table_name` for clarity

Comment (Th368MoJ) [reply splink/linker.py:178]: Completely agree.

**Pull request merged**